main.py

```diff
@@ -0,0 +1,80 @@
+import random
+
+
+class Unit:
+    def __init__(self, health, damage, defense, critical_power, percent_cp):
+        self._health = health
+        self._damage = damage
+        self._defense = defense
+        self._critical_power = critical_power
+        self._percent_cp = percent_cp
+
+    def __str__(self):
+        characteristic = f"Health: {self._health}, damage: {self._damage}, defense: {self._defense}, " \
+                         f"critical power: {self._critical_power}, percent critical power: {self._percent_cp}%"
+        return characteristic
+
+    def attack(self, unit, defense):
+        damage = self._damage
+        if random.random() < (self._percent_cp / 100):
```

Comment:
for simplication you can use randint(1, 100) and avoid dividing on 100, this will make your code much simplier